gpytorch/likelihoods/bernoulli_likelihood.py

```diff
@@ -56,3 +56,11 @@ def variational_log_probability(self, latent_func, target):
         samples = latent_func.rsample(torch.Size([num_samples])).view(-1)
         target = target.unsqueeze(0).repeat(num_samples, 1).view(-1)
         return log_normal_cdf(samples.mul(target)).sum().div(num_samples)
+
+    def pyro_sample_y(self, variational_dist_f, y_obs, sample_shape, name_prefix=""):
+        import pyro
+
+        f_samples = variational_dist_f(sample_shape)
+        y_prob_samples = torch.distributions.Normal(0, 1).cdf(f_samples)
+        y_dist = pyro.distributions.Bernoulli(y_prob_samples)
+        pyro.sample(name_prefix + "._training_labels", y_dist.independent(1), obs=y_obs)
```

Comment:
There's probably a way that we could consolidate `pyro_sample_y` and `variational_log_prob`. For example, lines 63-65 are the same for both. At this point, the use has to define 2 very similar functions for each likelihood.

Perhaps the user could define a `forward_samples` function that takes in samples from the MVN and returns a batch of likelihood samples?

Reply:
Then `variatioanl_log_prob` and `pyro_sample_y` would both call these functions.

Reply:
We just discussed this via slack, and it's not so simple since the two methods actually diverge right after sampling from the latent function distribution.